docs/resource/README.md

```diff
@@ -102,7 +102,117 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+yield different results for different services on the same host or namespace. When no namespaces or equivalent fields
+are available, the prefix MUST then be `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`.
+
+Users are encouraged to explicitly set the `service.instance.id` themselves, given they are in the best position
+to determine what's a service instance. For instance, in some cases, all containers in a given pod might be part
+of the same service, while in other cases, each container would independently generate their own telemetry. When
+it's not possible or desirable to explicitly generate a `service.instance.id`, users are encouraged to set set
+enough information via their workload automation, such as Helm. In that case, SDKs would have enough information to
+generate a `service.instance.id`. Future versions of the OpenTelemetry Operator might also have the ability to
+generate the `service.instance.id` and set container-specific labels, given it is in a position to know all that
+metadata. When SDKs don't have sufficient information to generate a stable UUID, a random one is generated.
+
+In more concrete terms: users running their services on platforms such as Kubernetes are encouraged to explicitly
+set the `service.instance.id` using their existing automation, or set a value that can be used for a
+consistent value, such as `container.id`. Similarly, users of application servers such as `unicorn`
+are encouraged to set the `service.instance.id` on a per-worker basis. SDKs MUST use the following algorithm when
+generating `service.instance.id`:
+
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this MUST take priority over generated IDs.
+- When any of the below combinations of resource attribute are provided, they MUST be used as the input
+  for generating a UUID v5 following the prefix mentioned above. The values within each combination MUST be separated with dots:
+  * `container.id`, resulting in the input `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${container.id}`,
+  possibly without the namespace.
+  * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`, resulting in the input
+  `${telemetry.sdk.name}.${telemetry.sdk.language}.${k8s.namespace.name}.${service.name}.${k8s.pod.name}.${k8s.container.name}`. In this case,
+  the namespace MUST be used.
+  * `host.id`, resulting in the input `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${host.id}`,
+  possibly without the namespace.
+- When the SDK is running in an environment where a `/etc/machine-id`
+  (see [MACHINE-ID(5)](https://www.freedesktop.org/software/systemd/man/machine-id.html))
+  is available, the machine-id should be used in the input for generating a UUID v5 (possibly without the namespace):
```

Comment:
```suggestion
  is available, the machine-id SHOULD be used in the input for generating a UUID v5 (possibly without the namespace):
```
Do we want to make this normative SHOULD?

Also: What does "possibly without the namespace" mean here? Does it mean it's optional in this case? If so, maybe change to "namespace is optional" instead? I feel right is a bit unclear what that means for SDK authors.